developer_docs/ansible_dev_setup.asciidoc

```diff
@@ -0,0 +1,142 @@
+= Ansible devel setup
+:toc: right
+:toclevels: 5
+
+[[prerequisites]]
+== Prerequisites
+
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/foreman_dev_setup.asciidoc[Foreman devel setup]
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/smart_proxy_dev_setup.asciidoc[Smart Proxy devel setup]
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/remote_execution_dev_setup.asciidoc[Remote Execution devel setup]
+* Installed https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html[Ansible]
+* Running machine that can be connected to via SSH (key authentication is best)
+
+[[foreman-plugin]]
+== Foreman plugin
+
+Download https://github.com/theforeman/foreman_ansible[foreman_ansible] plugin
+[source, bash]
+....
+git clone https://github.com/theforeman/foreman_ansible.git
+....
+
+Enable `foreman_ansible` plugin
+[source, ruby]
+....
+# bundler.d/ansible.local.rb
+
+gem 'foreman_ansible', path: '/path/to/foreman_ansible'
+....
+
+Install plugin dependencies & migrate database
+[source, bash]
+....
+bundle install
+npm install
+
+bundle exec rails db:migrate
+....
+
+
+[[smart-proxy-plugin]]
+== Smart Proxy plugin
+Add plugin gems
+
+[source, bash]
+....
+# /path/to/smart-proxy/bundler.d/ansible.local.rb
+
+gem 'smart_proxy_ansible', git: 'https://github.com/theforeman/smart_proxy_ansible.git'
+....
+
+Install gems
+[source, bash]
+....
+bundle install
+....
+
+Enable feature
+
+[source, yaml]
+....
+# config/settings.d/ansible.yml
+---
+:enabled: true
+....
+
+See https://github.com/theforeman/smart_proxy_ansible/blob/master/settings.d/ansible.yml.example[ansible.yml.example] file for full configuration.
+
+Run all services
+[source, bash]
+....
+bundle exec foreman start
+bundle exec rake dynflow:executor
+
+# From Smart Proxy folder
+bundle exec bin/smart-proxy
+# => [I] Successfully initialized 'ansible'
+....
+
+[[foreman-collection]]
+== Foreman collection
+
+`theforeman.foreman` collection contains Ansible Modules to manage Foreman and Katello installations.
+
+Install https://docs.ansible.com/ansible/latest/collections/theforeman/foreman/index.html[Foreman Collection]
+
+[source, bash]
+....
+ansible-galaxy collection install theforeman.foreman
+....
+
+Create symbolic link to folder where you have your collections
+
+[source, bash]
+....
+sudo ln -s /home/<your-account>/.ansible/collections/ansible_collections/ /usr/share/ansible/collections/ansible_collections
+....
+
+[[foreman-callback]]
+== Foreman callback
+This https://docs.ansible.com/ansible/latest/collections/theforeman/foreman/foreman_callback.html[callback] will report facts and task events to Foreman
+
+[source, bash]
+....
+# /etc/ansible/ansible.cfg
+
+[defaults]
+callbacks_enabled = foreman
+callback_plugins = /usr/share/ansible/collections/ansible_collections/theforeman/foreman/plugins/callback
```

Comment:
Adding here these two lines didn't work for me.
However adding this line instead works:
`callback_whitelist = theforeman.foreman.foreman`